eng/Versions.props

```diff
@@ -42,8 +42,8 @@
   </ItemGroup>
   <PropertyGroup>
     <!-- For source generator support we need to target a pinned version in order to be able to run on older versions of Roslyn -->
-    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>3.9.0</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
-    <MicrosoftCodeAnalysisVersion>3.9.0</MicrosoftCodeAnalysisVersion>
+    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.0.0-3.final</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
```

Comment:
This means the source generators will only work with the latest Roslyn version. Thus anyone using VS 2019 won't be able to use the source generators.

Reply:
I did wonder if there was some nuance to this based on the comment on line 44. I figured I'd push up the initial change, and then see what the impact/suggested resolution from the maintainers would be.

Reply:
We are working on a "multi-targeting" story for our source generators given https://github.com/dotnet/runtime/issues/56702#issuecomment-895578440. So maybe this will have to wait until we get the multi-targeting in place.